client/v2/indexer/searchForAssets.go

```diff
@@ -7,45 +7,54 @@ import (
 	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
 )
 
+// SearchForAssets /v2/assets
+// Search for assets.
 type SearchForAssets struct {
 	c *Client
 	p models.SearchForAssetsParams
 }
 
+// NextToken the next page of results. Use the next token provided by the previous
+// results.
 func (s *SearchForAssets) NextToken(nextToken string) *SearchForAssets {
 	s.p.NextToken = nextToken
 	return s
 }
 
+// Limit maximum number of results to return.
 func (s *SearchForAssets) Limit(lim uint64) *SearchForAssets {
 	s.p.Limit = lim
 	return s
 }
 
+// Creator filter just assets with the given creator address.
 func (s *SearchForAssets) Creator(creator string) *SearchForAssets {
 	s.p.Creator = creator
 	return s
 }
 
+// Name filter just assets with the given name.
 func (s *SearchForAssets) Name(name string) *SearchForAssets {
 	s.p.Name = name
 	return s
 }
 
+// Unit filter just assets with the given unit.
 func (s *SearchForAssets) Unit(unit string) *SearchForAssets {
 	s.p.Unit = unit
 	return s
 }
 
+// AssetID asset ID
 func (s *SearchForAssets) AssetID(id uint64) *SearchForAssets {
 	s.p.AssetId = id
 	return s
 }
 
-func (s *SearchForAssets) Do(ctx context.Context, headers ...*common.Header) (validRound uint64, result []models.Asset, err error) {
-	var response models.AssetsResponse
-	err = s.c.get(ctx, &response, "/v2/assets", s.p, headers)
-	validRound = response.CurrentRound
-	result = response.Assets
+// Do performs HTTP request
+func (s *SearchForAssets) Do(ctx context.Context,
+	headers ...*common.Header) (response models.AssetsResponse, err error) {
```

Comment:
We probably need to keep the old method signature... dev-rel has been cranking out example code that might need to be updated if we make a change like this.

The the tests could be fixed by construct the `AssetsResponse` object from `validRound` / `[]Asset`.

Let me get back to you on this.

Reply:
There is a bug in the old signature: it leaves out NextToken (see ApplicationsResponse below). 
The signature will change anyway, and all the usages will need to be updated. 

old signature:
`
func (s *SearchForAssets) Do(ctx context.Context, headers ...*common.Header) (validRound uint64, result []models.Asset, err error) {		`



```

type ApplicationsResponse struct {
	Applications []Application `json:"applications,omitempty"`

	// CurrentRound round at which the results were computed.
	CurrentRound uint64 `json:"current-round,omitempty"`

	// NextToken used for pagination, when making another request provide this token
	// with the next parameter.
	NextToken string `json:"next-token,omitempty"`
}

```

Reply:
Oh, nice catch. Thanks for the explanation